### DISCUSSION TITLE
Might we have lua 5.4 please?

### DISCUSSION THREAD

**Discussion by LinuxOnTheDesktop** [brndnmtthws/conky#1625] [Q&A] [answered]
Or at least instructions for how to get Conky to use lua5.4?

Cf. [this on the issue tracker](https://github.com/brndnmtthws/conky/issues/116), which was about getting lua5.3 working with Conky (and change that seemed to be accomplished via a change in Conky's code).

**Comment by brndnmtthws** [comment, marked as answer]
It should work with 5.4, what problem are you experiencing? Currently 5.3 or newer is required: https://github.com/brndnmtthws/conky/blob/ae733352f17bade39febbaf8da86b55d564cd109/cmake/ConkyPlatformChecks.cmake#L448

For `find_package`, `"5.3"` should also permit 5.4. See https://cmake.org/cmake/help/latest/command/find_package.html#basic-signature.

We're also using 5.4 for the nix flake: https://github.com/brndnmtthws/conky/blob/ae733352f17bade39febbaf8da86b55d564cd109/flake.nix#L70

**Comment by LinuxOnTheDesktop** [comment]
Ah: I had installed Lua 5.4, symlinked /usr/bin/lua to version 5.4, and then ran Conky - and expected Conky to use 5.4. You make me realise that I need to rebuild Conky. Thank you for the help.

**Comment by LinuxOnTheDesktop** [comment]
Sorry to bang on, but I rebuilt Conky, with Lua 5.4 installed, and Conky still uses Lua 5.3. I gather that I must supply some new argument to Cmake; at present i do `ccmake ..`. I know you supplied a link to documentation but I am afraid that I am having trouble understanding that documentation.

**Reply by brndnmtthws** [reply]
Try setting `LUA_FIND_VERSION=5.4`, per https://cmake.org/cmake/help/latest/command/find_package.html#package-file-interface-variables.

**Comment by LinuxOnTheDesktop** [comment]
Sorry, but like this?

```
cmake \
		-LUA_FIND_VERSION=5.4
```

But, ah; that doesn't work. Do I want the following?

```
$ export LUA_FIND_VERSION=5.4
$ cmake [arguments that I was using already]
```
